src/matcha_ml/services/analytics_service.py

```diff
@@ -27,12 +27,14 @@ class AnalyticsEvent(str, Enum):
 
 
 def execute_analytics_event(
-    func: Callable, *args, **kwargs
+    func: Callable, *args: dict, **kwargs: dict  # type: ignore
```

Comment:
not sure what's going on with the type hints here, but should *args really be a dict not a list?